### PR TITLE
Remove beacon from certificate

### LINKED ISSUE #1958: Remove `beacon` field in certificate
## Why
Remove the `beacon` field from the certificate which is optional for the client since distribution `2430`.

## What
Remove the `beacon` field from the certificate by following the `era_deprecate!("Remove immutable_file_number");` in the code

## How

- [x] Remove the `Beacon` field from the certificate 
- [ ] Assess if we need to recompute the certificate chain
- [ ] Announce breaking change in the next distribution

### PR DESCRIPTION
**Pull request by Alenar**
## Content

This PR remove the `beacon` field from the `CertificateMessage` and `CertificateListMessage` as this field was supersede by the `signed_entity_type` field, field that is mandatory since distribution [`2418.1`](https://github.com/input-output-hk/mithril/releases/tag/2418.1).

This allow to remove the `immutable_file_number` plus the conversion to cardano db beacon on the `CertificateMetadata` entity and the `CertificateRecord`, those only existed to recover the beacon when building a message.

> [!WARNING]
> 
> Nodes from distribution below [`2430`](https://github.com/input-output-hk/mithril/releases/tag/2430.0) are not compatible with this change.

## Pre-submit checklist

- Branch
  - [ ] Tests are provided (if possible)
  - [x] Crates versions are updated (if relevant)
  - [x] CHANGELOG file is updated (if relevant)
  - [x] Commit sequence broadly makes sense
  - [x] Key commits have useful messages
- PR
  - [x] No clippy warnings in the CI
  - [x] Self-reviewed the diff
  - [x] Useful pull request description
  - [x] Reviewer requested

## Comments

This changes should not induce a recompute of the certificates hashes as the immutable file number was not used in the certificate hash computation (the epoch and the network are still used).

## Issue(s)

Closes #1958

### CODE REVIEW

**Review by jpraynaud — APPROVED** [review]
LGTM :+1:

**Review by sfauvel — APPROVED** [review]
LGTM

File: mithril-common/src/messages/certificate_list.rs [hunk 1]
```diff
@@ -160,25 +153,54 @@ impl Debug for CertificateListItemMessage {
...
+            },
+            "signed_message": "signed_message",
+            "aggregate_verification_key": "aggregate_verification_key"
+        }]"#;
+
+    type CertificateListMessageUntilV0_1_32 = Vec<CertificateListItemMessageUntilV0_1_32>;
 
     #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-    struct CertificateListItemMessagePreviousVersion {
+    pub struct CertificateListItemMessageUntilV0_1_32 {
```
Comment (dlachaume) [inline mithril-common/src/messages/certificate_list.rs:190]: This struct doesn't need to be public.
```suggestion
    struct CertificateListItemMessageUntilV0_1_32 {
```

**Review by dlachaume — APPROVED** [review]
LGTM 🚀

**Pull request merged**